File: Fourth Semester - Design and Programming Environments/C#/ProiectC#/model/Agent.cs
namespace TravelAgency.model;

public class Agent: Entity<int>
{
    private string _name;
    private string _username;
    private string _password;

    public Agent(string name, string username, string password)
    {
        this._name = name;
        this._username = username;
        this._password = password;
    }

    public string Name
    {
        get => _name;
        set => _name = value;
    }

    public string Username
    {
        get => _username;
        set => _username = value;
    }

    public string Password
    {
        get => _password;
        set => _password = value;
    }
    
    public override string ToString()
    {
        return $"Agent Name: {_name}, Username: {_username}, Password: {_password}";
    }

    public override bool Equals(object? obj)
    {
        if (obj == null || GetType() != obj.GetType())
            return false;
        Agent other = (Agent)obj;
        return _name == other._name && _username == other._username && _password == other._password;
    }

    public override int GetHashCode()
    {
        return HashCode.Combine(_name, _username, _password);
    }
        
}

File: Fourth Semester - Design and Programming Environments/Java/ProiectJavaNetworking/TravelModel/src/java/org/example/model/Agent.java
package org.example.model;

import java.util.Objects;

public class Agent extends Entity<Integer> {
    private String name;
    private String username;
    private String password;

    public Agent(String agentName, String username, String password) {
        this.name = agentName;
        this.username = username;
        this.password = password;
    }

    public String getName() { return name; }

    public void setName(String name) { this.name = name;}

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    @Override
    public String toString() {
        return "Agent{" +
                "name='" + name + '\'' +
                ", username='" + username + '\'' +
                ", password='" + password + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if(!(obj instanceof Agent agent)) {
            return false;
        }
        return getName().equals(agent.getName()) && getUsername().equals(agent.getUsername()) && getPassword().equals(agent.getPassword());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getName(), getUsername(), getPassword());
    }
}
